src/w3d/lib/coord.h

```diff
@@ -22,104 +22,66 @@ using GameMath::Sqrt;
 class Coord2D
 {
 public:
-    Coord2D() : x(0.0f), y(0.0f) {}
-    Coord2D(float x_val, float y_val) : x(x_val), y(y_val) {}
-    float Length() { return float(Sqrt(float(float(x * x) + float(y * y)))); }
+    float Length() { return Sqrt(x * x + y * y); }
 
-    Coord2D &operator+=(const Coord2D &rhs)
+    void Normalize()
     {
-        x += rhs.x;
-        y += rhs.y;
+        float len = Length();
 
-        return *this;
+        if (len != 0.0f) {
+            x /= len;
+            y /= len;
+        }
     }
 
-    Coord2D &operator-=(const Coord2D &rhs)
+    float To_Angle()
     {
-        x -= rhs.x;
-        y -= rhs.y;
+        float len = Length();
 
-        return *this;
-    }
-
-    Coord2D &operator/=(const Coord2D &rhs)
-    {
-        x /= rhs.x;
-        y /= rhs.y;
-
-        return *this;
-    }
-
-    Coord2D &operator*=(const Coord2D &rhs)
-    {
-        x *= rhs.x;
-        y *= rhs.y;
+        if (len == 0.0f) {
+            return 0.0f;
+        }
 
-        return *this;
-    }
+        float f = x / len;
 
-    friend Coord2D operator+(Coord2D lhs, const Coord2D &rhs)
-    {
-        lhs += rhs;
+        if (f >= -1.0f) {
+            if (f > 1.0f)
+                f = 1.0f;
+        } else {
+            f = -1.0f;
+        }
 
-        return lhs;
+        if (y >= 0.0f) {
+            return GameMath::Acos(f);
+        } else {
+            return -GameMath::Acos(f);
+        }
     }
 
-    friend Coord2D operator-(Coord2D lhs, const Coord2D &rhs)
```

Comment:
Revert

Reply:
I added operators based on what Vector2 and Vector3 have and used them where appropriate.